File: algorithms/1_easy/M/01266_minimum_time_visiting_all_points/solution.cpp
class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int cost{0};
        const size_t len{points.size()};

        for (size_t i{1}; i < len; ++i) {
            cost += std::max(abs(points[i][0] - points[i - 1][0]), abs(points[i][1] - points[i - 1][1]));
        }

        return cost;
    }
};

File: algorithms/1_easy/M/01266_minimum_time_visiting_all_points/solution.rs
impl Solution {
    pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
        points.windows(2).map(|w| (w[1][0] - w[0][0]).abs().max((w[1][1] - w[0][1]).abs())).sum::<i32>()
    }
}
